diff -aur feh-pristine/src/signals.c feh-new/src/signals.c
--- feh-pristine/src/signals.c	2015-07-04 17:07:58.609892317 +0100
+++ feh-new/src/signals.c	2015-07-04 17:30:06.931235288 +0100
@@ -68,7 +68,6 @@
 void feh_handle_signal(int signo)
 {
 	winwidget winwid;
-	int i;
 
 	switch (signo) {
 		case SIGALRM:
@@ -85,15 +84,21 @@
 
 	winwid = winwidget_get_first_window_of_type(WIN_TYPE_SLIDESHOW);
 
-	if (winwid) {
-		if (signo == SIGUSR1)
-			slideshow_change_image(winwid, SLIDE_NEXT, 1);
-		else if (signo == SIGUSR2)
-			slideshow_change_image(winwid, SLIDE_PREV, 1);
-	} else if (opt.multiwindow) {
-		for (i = window_num - 1; i >= 0; i--)
-			feh_reload_image(windows[i], 0, 0);
-	}
+  if(!winwid)
+    winwid = winwidget_get_first_window_of_type(WIN_TYPE_UNSET);
+
+  if(!winwid)
+    winwid = winwidget_get_first_window_of_type(WIN_TYPE_SINGLE);
+
+  if(!winwid)
+    winwid = winwidget_get_first_window_of_type(WIN_TYPE_THUMBNAIL);
+
+  if(!winwid)
+    winwid = winwidget_get_first_window_of_type(WIN_TYPE_THUMBNAIL_VIEWER);
+
+  if(signo == SIGUSR1 && winwid){
+    feh_reload_image(winwid, 0, 0);
+  }
 
 	return;
 }
